src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -96,6 +98,20 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     }
   }
 
+  @Override
+  public ResponseEntity<SearchMetadataModel> upsertSearchMetadata(
+      @PathVariable("id") UUID id,
+      @Valid @RequestBody SearchMetadataRequest searchMetadataRequest) {
+    try {
+      SearchMetadataModel searchMetadataModel = new SearchMetadataModel();
+      searchMetadataModel.setMetadataSummary(
+          "Upserted search metadata for snapshot " + id.toString());
+      return new ResponseEntity<>(searchMetadataModel, HttpStatus.OK);
+    } catch (Exception e) {
```

Comment:
Do our controller endpoints normally have a top level handler for exceptions, vs letting exceptions get thrown? Either way the `GlobalExceptionHandler` will be invoked and convert the exception into a `500` response. I guess the question here is if it's useful to have a more specific error message here or not.

Reply:
We have other types of `Exception`s in other endpoints, including `IamForbiddenException` and `ValidationException`, and I thought we might have one of these other types of exceptions thrown, hence why I added a generic `Exception` section. Perhaps this was too premature, however.